### PR TITLE
NAS-112256 / 21.10 / TrueNAS R20B Enclosure Management

### PR DESCRIPTION
**Pull request by darkfiberiru**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/enclosure_/map.py [hunk 1]
```diff
@@ -96,7 +96,7 @@
             MappingSlot(0, 15, False),
         ]),
     ]),
-    ProductMapping(re.compile(r"TRUENAS-R20$"), [
+    ProductMapping(re.compile(r"TRUENAS-R20B?$"), [
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/enclosure_/map.py:99]: Should this not be `TRUENAS-R20?$`?

**Review by yocalebo — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/enclosure_/map.py (hunk 1 above)
Comment (darkfiberiru) [reply src/middlewared/middlewared/plugins/enclosure_/map.py:99]: No R20B? means zero or more B? so Matches R20 and R20B?

Comment (darkfiberiru) [reply src/middlewared/middlewared/plugins/enclosure_/map.py:99]: Hmm to many question marks

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/enclosure_/map.py:99]: I see now that `R20B` is the same as `R20`. Can we add a simple comment inside this mapping saying `R20` and `R20B` use same chassis, please?

Comment (darkfiberiru) [reply src/middlewared/middlewared/plugins/enclosure_/map.py:100]: Good?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/enclosure_/map.py:100]: perfect

**Review by yocalebo — APPROVED** [review]

File: src/middlewared/middlewared/plugins/enclosure.py [hunk 2]
```diff
@@ -47,6 +47,7 @@
 ES24_REGEX = re.compile(r"(ECStream|iX) 4024J")
 ES24F_REGEX = re.compile(r"(ECStream|iX) 2024J([ps])")
 MINI_REGEX = re.compile(r"(TRUE|FREE)NAS-MINI")
+R20_VARIANT = ["TRUENAS-R20", "TRUENAS-R20A","TRUENAS-R20B"]
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/enclosure.py:50]: flake8 funness 😄  need a space after end `,`

File: src/middlewared/middlewared/plugins/enclosure_/map.py [hunk 3]
```diff
@@ -96,7 +96,8 @@
             MappingSlot(0, 15, False),
         ]),
     ]),
-    ProductMapping(re.compile(r"TRUENAS-R20$"), [
+    #R20 and R20B share chassis and mapping
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/enclosure_/map.py:99]: flake8 funness 😄 comments need to start like `# ` (a space after the pound-sign)

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Comment by darkfiberiru** [comment]
Fixed flake complaints

**Review by yocalebo — APPROVED** [review]

**Pull request merged**